### PR TITLE
PXP-7942 Feat/newer react

### PR DESCRIPTION
**Pull request by mfshao**
Jira Ticket: [PXP-7942](https://ctds-planx.atlassian.net/browse/PXP-7942)

Update React beyond 16.8.6 to latest 16.x version

Rewrite some logic to get rid of deprecated React component lifecycle hooks

Update `core-js` from 2 to 3

Get rid of `@babel/polyfill` and use `@babel/preset-env` for applying polyfills based on usage

Updated eslint and linter config, and adopted eslint suggestions

Deployed in https://mingfei.planx-pla.net

### Breaking Changes
- Rewrite some logic to get rid of deprecated React component lifecycle hooks
- Removed `react-slick` and `slick-carousel`, use `antd` Carousel component instead

### Improvements
- Get rid of `@babel/polyfill` and use `@babel/preset-env` for applying polyfills based on usage

### Dependency updates
- Update React beyond 16.8.6 to latest 16.x version
- Update core-js from 2 to 3
- Updated eslint version and linter config

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
+  },
   root: true,
   env: {
     browser: true,
     es6: true,
     jest: true,
   },
+  plugins: [
+    'react',
+    'jsx-a11y',
```
Comment (williamhaley) [inline .eslintrc.js:21]: 💯

File: src/DataDictionary/graph/GraphDrawer/GraphDrawer.test.jsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import { mount } from 'enzyme';
-import GraphDrawer from '../GraphDrawer/GraphDrawer';
+import GraphDrawer from './GraphDrawer';
```
Comment (ZakirG) [inline src/DataDictionary/graph/GraphDrawer/GraphDrawer.test.jsx:3]: Why this change? Has the location of GraphDrawer moved?

Comment (mfshao) [reply src/DataDictionary/graph/GraphDrawer/GraphDrawer.test.jsx:3]: no, they are the same, both of those files are in `/GraphDrawer`

Comment (ZakirG) [reply src/DataDictionary/graph/GraphDrawer/GraphDrawer.test.jsx:3]: ahhh i see hahaha

**Review by ZakirG — APPROVED** [review]
Looks solid to me. Most of the file changes are only semantic. It works well in Mingfei's dev env. It passes Jenkins and Travis.
